### ISSUE TITLE
Background color on pdf

### ISSUE THREAD

**Issue by kamalsamra** [wkhtmltopdf/wkhtmltopdf#3125]
I want background color on multiple pages of pdf so for that i use background color on body but if content goes to second page then second page have background color till content.

**Comment by kamalsamra** [comment]
I got a solution for this. Pankaj Jangid one of the wkhtmltopdf developer provide this solution to me.
For this we have to add css :

```
html {
    height: auto;
  }

  body {
    height: auto;
    background:#000;
  }

  p.content,.resumesheet {
    page-break-after: always;
  }

  p.spacer: {
    display: block;
  }
```
Here you find sample css and html file .


[sample.zip](https://github.com/wkhtmltopdf/wkhtmltopdf/files/492992/sample.zip)

**Comment by rmcsharry** [comment]
FYI if you have margins, they remain white.

**Comment by angelo-moreira** [comment]
This doesn't work if we are instead loading dynamic content with lots of p tags :(

**Comment by pouyamn** [comment]
I have struggled with wkhtmltopdf for a proper background image for many days, I even tried to use background-position to synchronize display of header footer and content backgrounds, but every approach I tried became a dead end after many attempts. Just use another tool to finish it up:
[pdftk](https://www.pdflabs.com/docs/how-to-add-headers-footers-watermarks-and-stamps-to-pdf/)
[py2pdf](http://pybrary.net/pyPdf/pythondoc-pyPdf.pdf.html#pyPdf.pdf.PageObject.mergePage-method)

**Comment by herrwalter** [comment]
So, I ran across the same issue and stumbled op un this problem/solution and spent a lot of hours which I might save someone else fixing this. 
In my case I got inspired by the example of "spacing". 

So in my case we have a table that gets repeated. Sometimes on 1 page, sometimes on multiple pages. By adding a "filler" element after the repeating element (in our case, the table). 
The filler got this styling: 
```
        position: fixed;
        display: block;
        min-height: 100%;
        width: 210mm;
        background: #f4f2f1;
        margin: 0;
        padding: 0;
```

Position fixed is key here, along with the min-height. If it's on the first page, it will blend in with the background of the repeated element.
